azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/ILeaseManager.java

```diff
@@ -41,58 +42,63 @@
 	 * Does the lease store exist?
 	 * 
 	 * @return true if it does, false if not
+	 * @throws ExceptionWithAction with action EventProcessorHostActionStrings.CHECKING_LEASE_STORE on error
 	 */
-    public Future<Boolean> leaseStoreExists();
+    public CompletableFuture<Boolean> leaseStoreExists();
 
     /**
      * Create the lease store if it does not exist, do nothing if it does exist.
      * 
-     * @return true if the lease store already exists or was created successfully, false if not
+     * @return Void
+     * @throws ExceptionWithAction with action EventProcessorHostActionStrings.CREATING_LEASE_STORE on error
      */
-    public Future<Boolean> createLeaseStoreIfNotExists();
+    public CompletableFuture<Void> createLeaseStoreIfNotExists();
     
     /**
      * Not used by EventProcessorHost, but a convenient function to have for testing.
      * 
      * @return true if the lease store was deleted successfully, false if not
      */
-    public Future<Boolean> deleteLeaseStore();
+    public CompletableFuture<Boolean> deleteLeaseStore();
 
     /**
      * Return the lease info for the specified partition. Can return null if no lease has been
      * created in the store for the specified partition.
      * 
      * @param partitionId  id of partition to get lease for
-     * @return  lease info for the partition, or null
+     * @return lease info for the partition, or null
+     * @throws ExceptionWithAction with action EventProcessorHostActionStrings.GETTING_LEASE on error
      */
-    public Future<Lease> getLease(String partitionId);
+    public CompletableFuture<Lease> getLease(String partitionId);
     
     /**
      * Return the lease info for all partitions.
      * 
      * A typical implementation could just call getLease() on all partitions.
      * 
-     * @return  Iterable list of lease info.
+     * @return  List of lease info.
      */
-    public Iterable<Future<Lease>> getAllLeases() throws Exception;
+    public List<CompletableFuture<Lease>> getAllLeases(String[] partitionIds);
```

Comment:
> List<CompletableFuture<Lease>> [](start = 10, length = 31)

Change the contract to CompletableFuture<List<Lease>> - which puts the GetBlobs()-like optimization in the face of the API.